opennsa/setup.py

```diff
@@ -108,9 +109,17 @@ def setupTLSContext(vc):
         if not os.path.isdir(vc[config.CERTIFICATE_DIR]):
             raise config.ConfigurationError(
                 'certdir value {} is not a directory'.format(vc[config.CERTIFICATE_DIR]))
-        from opennsa.opennsaTlsContext import opennsaTlsContext
-        ctx_factory = opennsaTlsContext(
-            vc[config.CERTIFICATE_DIR], vc[config.VERIFY_CERT])
+        if vc[config.KEY] and vc[config.CERTIFICATE]:
+            # enable client authentication even when not in TLS mode
+            log.msg('setup for client TLS mode with client authentication')
+            from opennsa.opennsaTlsContext import opennsa2WayTlsContext
+            ctx_factory = opennsa2WayTlsContext(
+                vc[config.KEY], vc[config.CERTIFICATE], vc[config.CERTIFICATE_DIR], vc[config.VERIFY_CERT])
```

Comment:
Just a thought, rather than doing this, why don't you do a if config.KEY and config.Certificate is set, then just set the TLS flag, as the first check, then let it fall through.  It seems like lines 103-105 and lines 115-117 are basically the same.  

Or am I misreading something in the logic? would forcing TLS to be on break something with that pattern?

Reply:
The TLS flag is used to control the server side, if enabled then KEY and CERTIFICATE must be present, this is checked in config.py, if you do not want a TLS enabled server then TLS is set to false but if you still want to do client authentication then KEY and CERTIFICATE must be set.

But this code could be simpler, looking at config.py you can see that CERTIFICATE_DIR is always set that check can be removed.

What remains is, as you suggest, a check if CERTIFICATE and KEY are set (with or without TLS set), we then want a opennsa2WayTlsContext, otherwise just use opennsaTlsContext.

Reply:
Okay, I see.  I missed the import type.  Disregard as you were.

Reply:
I added a commit that simplifies setupTLSContext() as you suggest.